porcupine/plugins/directory_tree.py

```diff
@@ -385,12 +385,25 @@ def _on_right_click(self, event: tkinter.Event[DirectoryTree]) -> str | None:
         self.tk.call("focus", self)
 
         item: str = self.identify_row(event.y)
```

Comment:
The `identify_row()` doesn't make sense when this is triggered with a key press, so it should be moved under `if event.keysym != "App"`. You can also delete `: str` while you're at it. It should no longer be needed with the latest version of mypy.

Reply:
Okay, here's the new change:
```suggestion
        # Behaviour for a mouse right-click
        if event.keysym != "App":
            item = self.identify_row(event.y)
```

PS I'm still figuring out how to use GitHub right, and I'm not 100% sure on where to go from here...

Reply:
It's ok to make a mess with Git and GitHub. I can help if you get stuck :)

Instead of checking this with `event.keysym`, we could pass a boolean into the function. Something like this (with lots of details omitted):

```
class DirectoryTree:
    def on_right_click(self, menu_key: bool, event):
        ...
        if menu_key:
            ...
        else:
            ...

tree.bind("<<RightClick>>", partial(tree._on_right_click, False), add=True)
tree.bind("<<MenuKey>>", partial(tree._on_right_click, True), add=True)
```

Now right-click will run `tree._on_right_click(False, event object)`, and the menu key will run `tree._on_right_click(True, event object)`.